src/Server/LanguageServer.cs

```diff
@@ -357,9 +343,14 @@ private async Task Initialize(CancellationToken token)
         {
             ClientSettings = request;
 
-            if (request.Trace == InitializeTrace.Verbose && MinimumLogLevel >= LogLevel.Information)
+            if (request.Trace == InitializeTrace.Verbose)
             {
-                MinimumLogLevel = LogLevel.Trace;
+                var loggerSettings = _serviceProvider.GetService<LanguageServerLoggerSettings>();
+
+                if (loggerSettings?.MinimumLogLevel >= LogLevel.Information)
+                {
+                    loggerSettings.MinimumLogLevel = LogLevel.Trace;
+                }
```

Comment:
Best would be if we'd also here could notify `IOptionsMonitor<LoggerFilterOptions>` that it has changed. Without a configuration file, we'd probably have to do something like this. https://stackoverflow.com/a/53077453

Reply:
We should probably look at a configuration integration.  My thought is something like `interface ILspConfiguration : IConfiguration` and then populate that some defaults, like the logging state, but also wrap up the configuration handlers so that they surface their data through it, along with the refresh tokens.

Reply:
There's a better way - I had to do something similar for KubeClient:

https://github.com/tintoy/dotnet-kube-client/tree/develop/src/KubeClient.Extensions.Configuration

Reply:
Specifically:

https://github.com/tintoy/dotnet-kube-client/blob/develop/src/KubeClient.Extensions.Configuration/ConfigMapConfigurationProvider.cs#L155

Reply:
Then:

https://github.com/tintoy/dotnet-kube-client/blob/develop/src/KubeClient.Extensions.Configuration/ConfigMapConfigurationProvider.cs#L191

Reply:
I solved it by registering an own `IOptionsMonitor<LoggerFilterOptions>`